hazelcast/src/main/java/com/hazelcast/map/impl/wan/WanMapUpdateEvent.java

```diff
@@ -131,13 +147,33 @@ public long getCreationTime() {
         return 0;
     }
 
+    @Nonnull
     @Override
     public String getServiceName() {
         return MapService.SERVICE_NAME;
     }
 
+    @Nonnull
     @Override
     public String getObjectName() {
         return mapName;
     }
+
+    @Nonnull
+    @Override
+    public WanEventType getEventType() {
+        return WanEventType.ADD_OR_UPDATE;
+    }
+
+    @Nullable
+    @Override
+    @SuppressWarnings("unchecked")
+    public EntryView<Object, Object> getEventObject() {
+        return EntryViews.toLazyEntryView((EntryView) entryView, serializationService);
```

Comment:
seems i couldn't get why we need to return an `EntryView`. What is the target use case? Since entry-view contains lots of metadata.

Reply:
No special reason, really. I wanted some lazy-deserialising key-value pair that was possibly already exposed as public API.

Reply:
In any case, I guess it might be beneficial for the user to get more metadata about the entry, right? So they can replicate data like expiration time and hits.

Reply:
same as above: what about using `getEventObject` in our own implementation?

Reply:
Good idea, I suggest that I include this in the follow-up PR where I reuse the `WanMapEntryView` class both for public API and for internal use.